python_modules/libraries/dagster-postgres/dagster_postgres/schedule_storage/schedule_storage.py

```diff
@@ -172,6 +176,43 @@ def _add_or_update_instigators_table(self, conn: Connection, state: InstigatorSt
             )
         )
 
+    def add_auto_materialize_asset_evaluations(
+        self,
+        evaluation_id: int,
+        asset_evaluations: Sequence[AutoMaterializeAssetEvaluation],
+    ):
+        if not asset_evaluations:
+            return
+
+        insert_stmt = db_dialects.postgresql.insert(AssetDaemonAssetEvaluationsTable).values(
+            [
+                {
+                    "evaluation_id": evaluation_id,
+                    "asset_key": evaluation.asset_key.to_string(),
+                    "asset_evaluation_body": serialize_value(evaluation),
+                    "num_requested": evaluation.num_requested,
+                    "num_skipped": evaluation.num_skipped,
+                    "num_discarded": evaluation.num_discarded,
+                }
+                for evaluation in asset_evaluations
+            ]
+        )
+        upsert_stmt = insert_stmt.on_conflict_do_update(
+            index_elements=[
+                AssetDaemonAssetEvaluationsTable.c.evaluation_id,
+                AssetDaemonAssetEvaluationsTable.c.asset_key,
+            ],
+            set_={
+                "asset_evaluation_body": insert_stmt.excluded.asset_evaluation_body,
+                "num_requested": insert_stmt.excluded.num_requested,
```

Comment:
do we actually need to update here? Are there cases where the second time we write an evaluation, it's different?